src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -136,18 +133,40 @@ private PropertyCustomization introspectCustomization(Property property, JsonbCo
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
         final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
-        if (introspector.isTransient(property)) {
-            builder.setJsonbTransient(true);
-            return builder.buildPropertyCustomization();
+        EnumSet<AnnotationTarget> transientInfo = introspector.getJsonbTransientCategorized(property);
+        if (transientInfo.size() != 0) {
+            if(transientInfo.contains(AnnotationTarget.GETTER)) {
+                builder.setReadTransient(true);
+            }
+            if (transientInfo.contains(AnnotationTarget.SETTER)) {
+                builder.setWriteTransient(true);
+            }
+            if (transientInfo.contains(AnnotationTarget.PROPERTY)) {
+                if(!transientInfo.contains(AnnotationTarget.GETTER)){
+                    builder.setReadTransient(true);
+                }
+                if(!transientInfo.contains(AnnotationTarget.SETTER)){
+                    builder.setWriteTransient(true);
+                }
+            }
         }
-        builder.setJsonReadName(introspector.getJsonbPropertyJsonReadName(property));
-        builder.setJsonWriteName(introspector.getJsonbPropertyJsonWriteName(property));
-        builder.setNillable(classModel.getClassCustomization().isNillable() || introspector.isPropertyNillable(property));
+
+        if(!builder.isReadTransient()){
```

Comment:
Should exception already be risen here if JsonbTransient collides with other annotations?